cmd/kubeadm/app/cmd/phases/workflow/runner.go

```diff
@@ -0,0 +1,380 @@
+// This action can be executed explicitly out, when it is necessary to get the RunnerContext
+// before actually executing Run, or implicitly when invoking Run.
+func (e *Runner) BuildContext() (RunnerContext, error) {
+	if e.ctx == nil && e.contextBuilder != nil {
+		var err error
+		if e.ctx, err = e.contextBuilder(); err != nil {
+			return nil, err
+		}
+	}
+
+	return e.ctx, nil
+}
+
+// Run the kubeadm composable kubeadm workflows.
+func (e *Runner) Run() error {
+	e.prepareForExecution()
+
+	// determine which phase should be run according to RunnerOptions
+	phaseRunFlags, err := e.computePhaseRunFlags()
+	if err != nil {
+		return err
+	}
+
+	// builds the runner context
+	var ctx RunnerContext
+	if ctx, err = e.BuildContext(); err != nil {
+		return err
+	}
+
+	err = e.visitAll(func(p *phaseRunner) error {
+		// if the phase should not be run, skip the phase.
+		if run, ok := phaseRunFlags[p.generatedName]; !run || !ok {
+			return nil
+		}
+
+		// If the phase defines a condition to be checked before executing the phase action.
+		if p.RunIf != nil {
+			// Check the condition and returns if the condition isn't satisfied (or fails)
+			ok, err := p.RunIf(ctx)
+			if err != nil {
+				return fmt.Errorf("error execution run condition for phase %s: %v", p.generatedName, err)
+			}
+
+			if !ok {
+				return nil
+			}
+		}
+
+		// Runs the phase action (if defined)
+		if p.Run != nil {
+			if err := p.Run(ctx); err != nil {
+				return fmt.Errorf("error execution phase %s: %v", p.generatedName, err)
+			}
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+// Help returns text with the list of phases included in the workflow.
+func (e *Runner) Help(cmdUse string) string {
+	e.prepareForExecution()
+
+	// computes the max length of for each phase use line
+	maxLength := 0
+	e.visitAll(func(p *phaseRunner) error {
+		if !p.Hidden {
+			length := len(p.use)
+			if maxLength < length {
+				maxLength = length
+			}
+		}
+		return nil
+	})
+
+	// prints the list of phases indented by level and formatted using the maxlength
+	// the list is enclosed in a mardown code block for ensuring better readability in the public web site
+	line := fmt.Sprintf("The %q command executes the following internal workflow:\n", cmdUse)
+	line += "```\n"
+	offset := 2
+	e.visitAll(func(p *phaseRunner) error {
+		if !p.Hidden {
+			padding := maxLength - len(p.use) + offset
+			line += strings.Repeat(" ", offset*p.level) // indentation
+			line += p.use                               // name + aliases
+			line += strings.Repeat(" ", padding)        // padding right up to max length (+ offset for spacing)
+			line += p.Short                             // phase short description
+			line += "\n"
+		}
+
+		return nil
+	})
+	line += "```"
+	return line
+}
+
+// BindToCommand bind the Runner to a cobra command by altering
+// command help, adding phase related flags and by adding phases subcommands
+func (e *Runner) BindToCommand(cmd *cobra.Command) {
```

Comment:
Please add a comment that order matters here and that this needs to be done at the end once your phases are all added.

Reply:
done